openmc/aggregate.py

```diff
@@ -0,0 +1,410 @@
+            bin is a 2-tuple of floats for 'energy' and 'energyout' filters
+            corresponding to the energy boundaries of the bin of interest.
+            The bin is a (x,y,z) 3-tuple for 'mesh' filters corresponding to
+            the mesh cell of interest.
+
+        Returns
+        -------
+        filter_index : Integral
+             The index in the Tally data array for this filter bin. For an
+             AggregateTally the filter bin index is always unity.
+
+        Raises
+        ------
+        ValueError
+            When the filter_bin is not part of the aggregated filter's bins
+
+        """
+
+        if filter_bin not in self.bins:
+            msg = 'Unable to get the bin index for AggregateFilter since ' \
+                  '"{0}" is not one of the bins'.format(filter_bin)
+            raise ValueError(msg)
+        else:
+            return 0
+
+    def get_pandas_dataframe(self, datasize, summary=None):
+        """Builds a Pandas DataFrame for the AggregateFilter's bins.
+
+        This method constructs a Pandas DataFrame object for the AggregateFilter
+        with columns annotated by filter bin information. This is a helper
+        method for the Tally.get_pandas_dataframe(...) method.
+
+        Parameters
+        ----------
+        datasize : Integral
+            The total number of bins in the tally corresponding to this filter
+        summary : None or Summary
+            An optional Summary object to be used to construct columns for
+            distribcell tally filters (default is None). NOTE: This parameter
+            is not used by the AggregateFilter and simply mirrors the method
+            signature for the CrossFilter.
+
+        Returns
+        -------
+        pandas.DataFrame
+            A Pandas DataFrame with columns of strings that characterize the
+            aggregatefilter's bins. Each entry in the DataFrame will include
+            one or more aggregation operations used to construct the
+            aggregatefilter's bins. The number of rows in the DataFrame is the
+            same as the total number of bins in the corresponding tally, with
+            the filter bins appropriately tiled to map to the corresponding
+            tally bins.
+
+        See also
+        --------
+        Tally.get_pandas_dataframe(), Filter.get_pandas_dataframe(),
+        CrossFilter.get_pandas_dataframe()
+
+        """
+
+        import pandas as pd
+
+        # Construct a sring representing the filter aggregation
+        aggregate_bin = '{0}('.format(self.aggregate_op)
+        aggregate_bin += ', '.join(map(str, self.bins)) + ')'
+
+        # Construct NumPy array of bin repeated for each element in dataframe
+        aggregate_bin_array = np.array([aggregate_bin])
+        aggregate_bin_array = np.repeat(aggregate_bin_array, datasize)
+
+        # Construct Pandas DataFrame for the AggregateFilter
+        df = pd.DataFrame({self.type : aggregate_bin_array})
```

Comment:
Instead of `self.type` here, I would use `self.aggregate_filter.type`. It seems to more natural to keep the column name the same rather than change it to "sum(filter_type)", e.g., in Out[18] of the notebook I would still expect to see "energy [MeV]". The bin name already tells me that it is a sum.